src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator/AbstractResxGenerator.cs

```diff
@@ -22,166 +23,164 @@
+                    var hasNotNullIfNotNull = compilation.GetOrCreateTypeByMetadataName("System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute") is not null;
+
+                    return new CompilationInformation(
+                        AssemblyName: compilation.AssemblyName,
+                        CodeLanguage: compilation.Language,
+                        SupportsNullable: SupportsNullable(compilation),
+                        HasAggressiveInlining: hasAggressiveInlining,
+                        HasNotNullIfNotNull: hasNotNullIfNotNull);
+                });
+            var resourceFilesToGenerateSource = resourceFiles.Combine(context.AnalyzerConfigOptionsProvider).Combine(compilationInformation).SelectMany(
+                static (resourceFileAndOptions, cancellationToken) =>
                 {
-                    var exceptionLines = ex.ToString().Split(new[] { Environment.NewLine }, StringSplitOptions.None);
-                    var text = string.Join("", exceptionLines.Select(line => "#error " + line + Environment.NewLine));
-                    var errorText = SourceText.From(text, Encoding.UTF8, SourceHashAlgorithm.Sha256);
-                    context.AddSource($"{Path.GetFileName(resourceFile.Path)}.Error", errorText);
-                }
-            }
-        }
-
-        private void ProcessResourceFile(GeneratorExecutionContext context, AdditionalText resourceFile)
-        {
-            var options = context.AnalyzerConfigOptions.GetOptions(resourceFile);
+                    var ((resourceFile, optionsProvider), compilationInfo) = resourceFileAndOptions;
+                    var options = optionsProvider.GetOptions(resourceFile);
+                    if (options.TryGetValue("build_metadata.AdditionalFiles.GenerateSource", out var generateSourceText)
+                        && !string.IsNullOrEmpty(generateSourceText)
+                        && generateSourceText != "unset"
+                        && !string.Equals(bool.TrueString, generateSourceText, StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Source generation is disabled for this resource file
+                        return Array.Empty<ResourceInformation>();
+                    }
 
-            if (options.TryGetValue("build_metadata.AdditionalFiles.GenerateSource", out var generateSourceText)
-                && !string.IsNullOrEmpty(generateSourceText)
-                && generateSourceText != "unset"
-                && !string.Equals(bool.TrueString, generateSourceText, StringComparison.OrdinalIgnoreCase))
-            {
-                // Source generation is disabled for this resource file
-                return;
-            }
+                    if (!optionsProvider.GlobalOptions.TryGetValue("build_property.RootNamespace", out var rootNamespace))
+                    {
+                        rootNamespace = compilationInfo.AssemblyName;
```

Comment:
similarly, i only see compilationInfo used here (And passed to the final ResourceInformation value).  Consider doing all the other work in a Select on the filtered `(resourceFile, optionsProvider)`.  Then finally combine that with compilationINfo only to get hte root namespace and add that in.

Reply:
Considering that `compilationInfo` only holds data that is _particularly_ unlikely to change from one compilation to the next, would we still expect the extraction would be beneficial?

Reply:
📝 AnalyzerConfigOptionsProvider is the input that changes most often for this step. The contents of this `SelectMany` represent the extraction of data from this value such that the impact on the pipeline stops after this step unless the change was meaningful.

Reply:
➡️ Keeping this portion for now, since it's a targeted extraction of data from `AnalyzerConfigOptionsProvider` for each resource